pgjdbc/src/main/java/org/postgresql/core/Utils.java

```diff
@@ -197,8 +151,7 @@ private static void doAppendEscapedIdentifier(Appendable sbuf, String value) thr
     try {
       sbuf.append('"');
 
-      for (int i = 0; i < value.length(); ++i) {
-        char ch = value.charAt(i);
+      for (char ch : value.toCharArray()) {
```

Comment:
I wonder if this change would result array allocation and extra cpu/memory consumption issues